xml_analyzer/calculator.py

```diff
@@ -1,63 +1,25 @@
-import xml.etree.ElementTree as Et
-from enum import Enum
-from .formula import Formula
 from .data_container import DataContainer
-
-
-class EquationTypes(Enum):
-    equation = 'equation'
-    inequation_gr_th = 'inequation_gr_th'
-    inequation_sm_th = 'inequation_sm_th'
-    inequation_gr_eq = 'inequation_gr_eq'
-    inequation_sm_eq = 'inequation_sm_eq'
+from .formula import Formula
 
 
 class Calculator:
-    def __init__(self, constraint_property: Et.Element, property_value_mapping: DataContainer):
-        self.constraint_property = constraint_property
-        self.property_value_mapping = property_value_mapping
+    def __init__(self, data: DataContainer):
+        self.data = data
+        self.formula = Formula(data.constraint_specification, data.get_variable_names())
 
     @staticmethod
-    def get_expression_type(expr: str) -> [EquationTypes, str]:
-
-        if '>=' in expr:
-            expr_type = EquationTypes.inequation_gr_eq
-            expr_sep = '>='
-        elif '<=' in expr:
-            expr_type = EquationTypes.inequation_sm_eq
-            expr_sep = '<='
-        elif '>' in expr:
-            expr_type = EquationTypes.inequation_gr_th
-            expr_sep = '>'
-        elif '<' in expr:
-            expr_type = EquationTypes.inequation_sm_th
-            expr_sep = '<'
-        elif '=' in expr:
-            expr_type = EquationTypes.equation
-            expr_sep = '='
-        else:
-            expr_type = 'undef'
-            expr_sep = 'undef'
-
-        return expr_type, expr_sep
-
-    def resolve_auto_calc_function(self, autocalc_method, type_name):
-        retval = None
+    def _resolve_auto_calc_functions(autocalc_method, values):
         if autocalc_method == 'autosum':
-            retval = self.do_autosum(type_name)
-
-        if autocalc_method == 'automult':
-            pass
+            return round(sum(values), 3)
 
-        return retval
+    def calculate(self) -> float:
+        for autocalc_method in self.data.mappings['autocalc']:
```

Comment:
handle string keys like 'autocalc' like a magic number. If the name changes you have to change it everywhere you used it

Reply:
Fixed

Reply:
Better solution with https://github.com/Squiddrone/VPParametricsSolver/pull/2/commits/56aed8eba9b4acceccf71949cee0658b3197752c